db/collection.go

```diff
@@ -453,3 +585,37 @@ func setNewInstanceID(t []byte) (core.InstanceID, []byte) {
 	}
 	return newID, patchedValue
 }
+
+func compileJSFunc(v []byte, args ...string) ([]byte, error) {
+	if v != nil && len(v) == 0 {
+		v = nil
+	}
+	if v == nil {
+		return nil, nil
+	}
```

Comment:
This could be simplified a bit, [since `len(nilSlice)==0`](https://play.golang.org/p/TFp5pb2ta_2).